kubernetes/kubernetes/src/main/java/org/arquillian/cube/kubernetes/impl/SessionManager.java

```diff
@@ -185,21 +202,102 @@ public void createEnvironment(Session session) {
     public void start() {
         Logger log = session.getLogger();
         log.status("Using Kubernetes at: " + client.getMasterUrl());
-        createNamespace(session);
+        createNamespace();
+        setupConsoleListener();
 
         addShutdownHook();
         try {
-            createEnvironment(session);
+            createEnvironment();
         } catch (Throwable t){
           removeShutdownHook();
           throw t;
         }
     }
 
+    private void addConsole(final String podName) {
+        if (watchersMap.containsKey(podName))
+            return;
+
+        String className = session.getCurrentClassName();
+        String methodName = session.getCurrentMethodName();
+        String fileName = consoleLogPath;
+
+        if (Strings.isNullOrEmpty(className))
+            className = "NOCLASS";
+        fileName += String.format("/%s", className);
+
+        if (Strings.isNotNullOrEmpty(methodName))
+            fileName += String.format("-%s", methodName);
+        fileName += String.format("-%s.log", podName);
+
+        try {
+            final FileOutputStream stream = new FileOutputStream(fileName);
+            LogWatch lw = client.pods().inNamespace(session.getNamespace()).withName(podName).watchLog(stream);
```

Comment:
How about if pod has multiple containers? I believe container name is optional if pod has only one container.

Again if we provide support for this we can get logs of container from pod.

Reply:
Yeah, I didn't think of pods with multiple containers... The easiest way would be to capture the logs of the first container, though I'm not sure it's the best thing to do... I'll think a bit more about it.

Reply:
I've updated the patch so that we get the logs for all containers within a pod. So, if a pod has 3 containers, we'll get 3 log files, with the container name being part of the filename.